commitizen/commands/commit.py

```diff
@@ -91,7 +91,11 @@ def __call__(self):
         signoff: bool = self.arguments.get("signoff")
 
         if signoff:
+            out.warn("signoff mechanic is deprecated, please use `cz commit -- -s` instead.")
```

Comment:
@woile Do you think we should add a list somewhere so we won't forget to deprecate stuff in the next release?

Reply:
@SCrocky In this implementation, if we're using the original `signoff` feature, we won't be able to use `extra_cli_args.`. Should we implement it as something like the following? (the warning is still needed)

```python
extra_cli_args = ""

if signoff:
    extra_cli_args = "-s "
```

Reply:
Ah, I see it. Thanks for that.

Would this work for you?

```python
        if signoff:
            out.warn("signoff mechanic is deprecated, please use `cz commit -- -s` instead.")
            extra_args = "-s " + self.arguments.get("extra_cli_args")
        else:
            extra_args = self.arguments.get("extra_cli_args")

        c = git.commit(m, extra_args=extra_args)
```

Reply:
I might try something like

```python
        extra_args = self.arguments.get("extra_cli_args", "")
        if signoff:
            out.warn("signoff mechanic is deprecated, please use `cz commit -- -s` instead.")
            extra_args += " -s"

        c = git.commit(m, extra_args=extra_args)
```

Reply:
But one thing I'm thinking is whether we should handle `cz commit -s -- -s`

Reply:
I tested it and doesn't seem to be an issue with the git command:
```sh
git commit -s -s -m "test commit" 
[master (root-commit) b864659] test commit
 1 file changed, 0 insertions(+), 0 deletions(-)
```

But it is a bit messy. Warning or error?

Reply:
If it doesn't cause an error, I'm good with either silently passing or warning.